src/libraries/Microsoft.Win32.SystemEvents/src/Microsoft/Win32/SystemEvents.cs

```diff
@@ -130,17 +130,12 @@ public static event EventHandler? DisplaySettingsChanged
         ///  Occurs before the thread that listens for system events is terminated.
         ///  Delegates will be invoked on the events thread.
         /// </summary>
+        [Obsolete("Callbacks are not guaranteed to run before process exits. Use AppDomain.ProcessExit instead.")]
```

Comment:
There is a lot more ceremony around obsoleting APIs in this repo. The process is described in: https://github.com/dotnet/runtime/blob/main/docs/project/list-of-diagnostics.md